kotlin-eclipse-core/src/org/jetbrains/kotlin/core/resolve/lang/java/structure/EclipseJavaElementUtil.java

```diff
@@ -234,7 +238,17 @@ private static boolean isKotlinClassFile(IClassFile classFile) {
         if (binaryClass == null) {
             return false;
         }
-        return true;
+        if (classFile instanceof ClassFile) {
```

Comment:
Can you provide an example of when this check is necessary?
`getKotlinBinaryClassOrClassFileContent` checks annotations of a binary file internally, so maybe we should fix it instead (in the compiler)

Reply:
Oh, wait, I found why this is happening. The behavior of `getKotlinBinaryClassOrClassFileContent` has recently [changed](https://github.com/JetBrains/kotlin/blob/master/compiler/frontend.java/src/org/jetbrains/kotlin/load/kotlin/VirtualFileKotlinClass.kt#L72), now I agree with your changes, thanks